### PR TITLE
Correction of alter strategy binary type bug + test

### PR DESCRIPTION
**Pull request by khalilTN**
When activating the "Alter" mode, waterline retrieve data, delete tables, recreate them and then reinsert data, if data contains buffers, executing cloneDeep() de lodash causes buffer to be converted to another type of object and then data loss. In this PR, correction of the bug and a test of the mode "alter".

### CODE REVIEW

**Comment by atiertant** [comment]
fix https://github.com/balderdashy/waterline/issues/996

**Comment by dmarcelino** [comment]
First impressions: the indentation is a bit off, most (all?) of waterline files use 2 spaces for tab size. Can you please fix it?

File: test/unit/adapter/strategy.alter.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+            }
+        };
+        connections = {
+            'test_alter': {
+                adapter: 'fake'
+            }
+        };
+        adapters = {fake: adapter};
+    });
+    it('should recover data', function (done) {this.timeout(10000);
```
Comment (dmarcelino) [inline test/unit/adapter/strategy.alter.js]: Style alert, can `this.timeout(10000);` be moved to the line below?

**Comment by khalilTN** [comment]
done

**Comment by dmarcelino** [comment]
Thanks @khalilTN, I think this is good! :white_check_mark: 

@devinivy, what do you think?

**Comment by dmarcelino** [comment]
@tjwebb, can you please take a look?

**Pull request merged**